suave/cstore/engine2.go

```diff
@@ -0,0 +1,283 @@
+package cstore
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	suave "github.com/ethereum/go-ethereum/suave/core"
+	"github.com/google/uuid"
+
+	"github.com/gorilla/mux"
+)
+
+type Engine2 struct {
+	storage ConfidentialStorageBackend
+	server  *http.Server
+
+	sessions     map[string]*TransactionalStore2
+	sessionsLock sync.Mutex
+}
+
+func NewEngine2(storage ConfidentialStorageBackend) *Engine2 {
+	a := &Engine2{
+		storage:  storage,
+		sessions: make(map[string]*TransactionalStore2),
+	}
+
+	r := mux.NewRouter()
+
+	r.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "Not found", http.StatusNotFound)
+	})
+
+	r.HandleFunc("/cstore/new", a.handleNewSession).Methods("POST")
+	r.HandleFunc("/cstore/{id}/record", a.handleNewRecord).Methods("POST")
+	r.HandleFunc("/cstore/{id}/record/{dataid}/{key}", a.handleRecordStorePost).Methods("POST")
+	r.HandleFunc("/cstore/{id}/record/{dataid}/{key}", a.handleRecordStoreGet).Methods("GET")
+	r.HandleFunc("/cstore/{id}/record/{dataid}", a.handleRecordGet).Methods("GET")
+	r.HandleFunc("/cstore/{id}/fetchByBlock", a.handleFetchByBlock).Methods("GET")
+	r.HandleFunc("/cstore/{id}/finalize", a.handleFinalizeSession).Methods("POST")
+
+	a.server = &http.Server{
+		Addr:    ":8080",
+		Handler: r,
+	}
+
+	return a
+}
+
+func (a *Engine2) Start() error {
+	log.Info("Server started on :8080")
+
+	go func() {
+		if err := a.server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Error("Error starting server: %s\n", err)
```

Comment:
I'm noting some inconsistency in capitalization in log messages.  I believe the convention in Go (and most languages) is not to capitalize log strings.  I would suggest doing the same here.

I'll flag the other locations in the PR.

Reply:
All the log prints are capitalised in Geth https://github.com/search?q=repo%3Aflashbots%2Fsuave-geth%20log.Print&type=code. Do you refer to error messages or prints on the console?